transport/tchannel.go

```diff
@@ -75,7 +81,7 @@ func NewTChannel(opts TChannelOptions) (Transport, error) {
 	}
 
 	callerName := opts.SourceService
-	if cn, ok := opts.TransportOpts["cn"]; ok {
+	if cn, ok := opts.TransportOpts["cn"]; ok && cn != "" {
```

Comment:
nit: should we keep this as is -- it might be we want to test what happens if the `cn` header is empty in a request.